src/main/java/redis/clients/jedis/JedisClusterConnectionHandler.java

```diff
@@ -29,12 +34,15 @@ public Jedis getConnectionFromNode(HostAndPort node) {
     return cache.getNodes();
   }
 
-  private void initializeSlotsCache(Set<HostAndPort> startNodes, GenericObjectPoolConfig poolConfig, String password) {
+  private void initializeSlotsCache(Set<HostAndPort> startNodes, GenericObjectPoolConfig poolConfig, String password, String clientName) {
     for (HostAndPort hostAndPort : startNodes) {
       Jedis jedis = new Jedis(hostAndPort.getHost(), hostAndPort.getPort());
       if (password != null) {
         jedis.auth(password);
       }
+      if (clientName != null) {
```

Comment:
This instance is for discovery and will be closed soon. Do we still want to apply client setname here?